### PR TITLE
Make ddl follow tt mechanism

### PR DESCRIPTION
**Pull request by longjiquan**
Signed-off-by: longjiquan <jiquan.long@zilliz.com>
issue: #19175

### CODE REVIEW

File: internal/rootcoord/step_executor.go [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+	for _, s := range selected {
+		bg.removeStepsInternal(s)
+	}
+	bg.mu.Unlock()
+
+	bg.process(selected)
+}
+
+func (bg *bgStepExecutor) scheduleLoop() {
+	bg.wg.Add(1)
```
Comment (czs007) [inline internal/rootcoord/step_executor.go:170]: should moved to  outside of scheduleLoop

File: internal/rootcoord/step_executor.go [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+	defer bg.wg.Done()
+
+	ticker := time.NewTicker(bg.interval)
```
Comment (czs007) [inline internal/rootcoord/step_executor.go:173]: should defer call ticker.Stop

File: internal/rootcoord/garbage_collector.go [hunk 3]
```diff
@@ -34,125 +27,84 @@ func newGarbageCollectorCtx(s *Core) *GarbageCollectorCtx {
...
+			physicalChannels: collMeta.PhysicalChannelNames,
+		},
+	})
+	redo.AddAsyncStep(&deleteCollectionMetaStep{
+		baseStep:     baseStep{core: c.s},
+		collectionID: collMeta.CollectionID,
+		ts:           collMeta.CreateTime,
+	})
+	// err is ignored since no sync steps will be executed.
+	_ = redo.Execute(context.Background())
```
Comment (czs007) [inline internal/rootcoord/garbage_collector.go:75]: may be we can log.Info this err ?

File: internal/rootcoord/ddl_ts_lock_manager.go [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+	"github.com/milvus-io/milvus/api/commonpb"
+	ms "github.com/milvus-io/milvus/internal/mq/msgstream"
+	"github.com/milvus-io/milvus/internal/proto/internalpb"
+
+	"go.uber.org/atomic"
+
+	"github.com/milvus-io/milvus/internal/metastore/model"
+)
+
+type DdlTsLockManager interface {
```
Comment (czs007) [inline internal/rootcoord/ddl_ts_lock_manager.go:16]: How about expose 
GetMinDdlTs()
Lock()
UnLock()
UpdateLastTs 
?

Comment (czs007) [reply internal/rootcoord/ddl_ts_lock_manager.go:16]: and remove NotifyCollectionGc and NotifyPartitionGc

File: internal/rootcoord/garbage_collector.go [hunk 5]
```diff
@@ -2,25 +2,18 @@ package rootcoord
...
 	ReDropCollection(collMeta *model.Collection, ts Timestamp)
 	RemoveCreatingCollection(collMeta *model.Collection)
 	ReDropPartition(pChannels []string, partition *model.Partition, ts Timestamp)
-	GcCollectionData(ctx context.Context, coll *model.Collection, ts typeutil.Timestamp) error
-	GcPartitionData(ctx context.Context, pChannels []string, partition *model.Partition, ts typeutil.Timestamp) error
+	GcCollectionData(ctx context.Context, coll *model.Collection, ts typeutil.Timestamp) (ddlTs Timestamp, err error)
+	GcPartitionData(ctx context.Context, pChannels []string, partition *model.Partition, ts typeutil.Timestamp) (ddlTs Timestamp, err error)
 }
 
 type GarbageCollectorCtx struct {
```
Comment (czs007) [inline internal/rootcoord/garbage_collector.go:19]: Ctx? a bit of confusing and misunderstanding

File: internal/rootcoord/garbage_collector.go [hunk 6]
```diff
@@ -34,125 +27,84 @@ func newGarbageCollectorCtx(s *Core) *GarbageCollectorCtx {
...
-	msgPack.Msgs = append(msgPack.Msgs, msg)
-	if err := c.s.chanTimeTick.broadcastDmlChannels(pChannels, &msgPack); err != nil {
-		return err
-	}
-
-	// TODO: remove this after gc can be notified by rpc. Without this tt, DropCollectionMsg cannot be seen by
-	// 		datanodes.
-	return c.s.chanTimeTick.sendTimeTickToChannel(pChannels, ts)
+func (c *GarbageCollectorCtx) GcPartitionData(ctx context.Context, pChannels []string, partition *model.Partition, ts typeutil.Timestamp) (ddlTs Timestamp, err error) {
+	return c.s.ddlTsLockManager.NotifyPartitionGc(ctx, pChannels, partition)
```
Comment (czs007) [inline internal/rootcoord/garbage_collector.go:109]: How about :
c.s.ddlTsLockManager.Lock()
allocate ts
do the NotfifyPartitionGC stuff as a method of GarbageCollector
c.s.ddlTsLockManager.updateLastTs(ts)
c.s.ddlTsLockManager.UnLock

File: internal/rootcoord/step_executor.go (hunk 1 above)
Comment (longjiquan) [reply internal/rootcoord/step_executor.go:170]: OK, I'll this.

File: internal/rootcoord/step_executor.go (hunk 2 above)
Comment (longjiquan) [reply internal/rootcoord/step_executor.go:173]: OK, I'll this.

File: internal/rootcoord/garbage_collector.go (hunk 3 above)
Comment (longjiquan) [reply internal/rootcoord/garbage_collector.go:75]: `redo.Execute` here won't return error, it always return `nil`.

File: internal/rootcoord/garbage_collector.go (hunk 5 above)
Comment (longjiquan) [reply internal/rootcoord/garbage_collector.go:19]: How about `bgGarbageCollector`?

File: internal/rootcoord/garbage_collector.go (hunk 6 above)
Comment (longjiquan) [reply internal/rootcoord/garbage_collector.go]: OK.

**Pull request merged**